rl4co/envs/routing/ovrp.py

```diff
@@ -0,0 +1,440 @@
+from typing import Optional
+
+import torch
+
+from tensordict.tensordict import TensorDict
+from torchrl.data import (
+    BoundedTensorSpec,
+    CompositeSpec,
+    UnboundedContinuousTensorSpec,
+    UnboundedDiscreteTensorSpec,
+)
+
+from rl4co.data.utils import load_npz_to_tensordict
+from rl4co.envs.common.base import RL4COEnvBase
+from rl4co.utils.ops import gather_by_index, get_open_tour_length
+from rl4co.utils.pylogger import get_pylogger
+
+log = get_pylogger(__name__)
+
+
+# From Kool et al. 2019, Hottung et al. 2022, Kim et al. 2023
+CAPACITIES = {
+    10: 20.0,
+    15: 25.0,
+    20: 30.0,
+    30: 33.0,
+    40: 37.0,
+    50: 40.0,
+    60: 43.0,
+    75: 45.0,
+    100: 50.0,
+    125: 55.0,
+    150: 60.0,
+    200: 70.0,
+    500: 100.0,
+    1000: 150.0,
+}
+
+
+class OVRPEnv(RL4COEnvBase):
+    """Capacitated Vehicle Routing Problem (CVRP) environment.
```

Comment:
Thank you for the suggestions and refinements ~ 
1): in the multi-task learning (MTL) case (my paper) the VRP environments are implemented together in a single MT environment. I am currently exploring ways to incorporate MTL into the RL4CO flowchart. It would be more convenient if we could directly integrate MTL with the existing single-task Envs in RL4CO instead of creating separate MTL Envs for each MT combination. As the first stage, the four VRP Envs have been included in RL4CO as individual VRP Envs.

2): Yes, the VRPB settings are slightly different from traditional settings. In this case, only one resource is permitted for each node, and the total demands carried by the vehicle may rise, while the maximum delivery capacity (at the depot) remains constant. This adjustment is made for ease of use in the same RL framework with a masking procedure. We may revise the settings to meet diverse real-world cases.

Reply:
1) I see! [This](https://github.com/FeiLiu36/MTNCO/blob/main/MTPOMO/POMO/VRPEnv.py) is the environment, right? A similar procedure can be adopted in RL4CO too. I guess the single environments without feature combinations could be added as well first for clarity, while for the combination, an environment as `AnyVRP` (as we are calling it) or `MTLVRP` 

2) I see! Do you have some references about the problem variant in which one resource can be transported from a node to another?

Reply:
1. Yes, you are right! I would suggest MTLVRP or MTVRP if it is possible~

2. The setting can be regarded as mixed VRPB, where a single resource can serve both as a linehaul and backhaul. This is one of the pioneering works on mixed VRPB (An investigation into a new class of vehicle routing problem with backhauls. Omega, 2002.). The recent revisions of mixed VRPB and more variants can be found in this review (Vehicle routing with backhauls: Review and research perspectives. Computers & Operations Research, 2018)